aep/general/0135/aep.md.j2

```diff
@@ -0,0 +1,314 @@
+comprehensive treatment: [AEP-164][aep-164].
+
+### Long-running delete
+
+Some resources take longer to delete than is reasonable for a
+regular API request. In this situation, the API **should** use a long-running
+operation instead. See [AEP-151][aep-151] for full details on long-running operations.
+
+{% tab oas -%}
+
+{% sample 'long_running_delete.oas.yaml', 'paths' %}
+
+- The response status code should be `202 Accepted` if the request was
+  accepted for later processing. When the request is processed it could still fail.   
+- The `response` field of the response body **must** be set to the appropriate return type if the RPC
+  was not long-running: `google.protobuf.Empty` for most Delete RPCs, or the
+  resource itself for soft delete ([AEP-164][aep-164]).
+- Both the `response_type` and `metadata_type` fields **must** be specified
+  (even if they are `google.protobuf.Empty`).
+
+{% tab proto -%}
+
+```proto
+rpc DeleteBook(DeleteBookRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    delete: "/v1/{name=publishers/*/books/*}"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "google.protobuf.Empty"
+    metadata_type: "OperationMetadata"
+  };
+}
+```
+
+- The response type **must** be set to the appropriate return type if the RPC
+  was not long-running: `google.protobuf.Empty` for most Delete RPCs, or the
+  resource itself for soft delete ([AEP-164][aep-164]).
+- Both the `response_type` and `metadata_type` fields **must** be specified
+  (even if they are `google.protobuf.Empty`).
+
+{% endtabs %}
+
+### Cascading delete
+
+Sometimes, it may be necessary for users to be able to delete a resource as
+well as all applicable child resources. However, since deletion is usually
+permanent, it is also important that users not do so accidentally, as
+reconstructing wiped-out child resources may be quite difficult.
+
+If an API allows deletion of a resource that may have child resources, the API
+**should** provide a `bool force` field on the request, which the user sets to
+explicitly opt in to a cascading delete.
+
+{% tab oas -%}
+
+{% sample 'cascading_delete.oas.yaml', 'paths' %}
+
+The API **must** fail with a `409 Conflict` error if the `force` field
+is `false` (or unset) and child resources are present.
+
+{% tab proto -%}
+
+```proto
+message DeletePublisherRequest {
+  // The name of the publisher to delete.
+  // Format: publishers/{publisher}
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/Publisher"
+    }];
+
+  // If set to true, any books from this publisher will also be deleted.
+  // (Otherwise, the request will only work if the publisher has no books.)
+  bool force = 2;
+}
+```
+
+The API **must** fail with a `FAILED_PRECONDITION` error if the `force` field
+is `false` (or unset) and child resources are present.
+
+{% endtabs %}
+
+### Protected delete
```

Comment:
do you think this guidance is needed here? feels like we could just consolidate this by just mentioning and linking to an etag section.

Reply:
That seems reasonable. I'll drop it for now -- if there are strong objections we can add it back.

Reply:
FWIW, I think the bit about protected delete is useful guidance (people may not think to use etags in that way) and it'd be fine to keep it.

Reply:
From live discussion with Yusuke:  harder for us as maintainers to keep stuff like this, but maybe useful for readers to have illustrative examples.  Yusuke is okay keeping this.

Another point:  the AEPs generally say standard method requests and responses can't have fields that aren't explicitly specified somewhere by the AEPs.  This persuaded Yusuke.